### PR TITLE
zone-setup cleanup/refactoring

### PR DESCRIPTION
**Pull request by jgallagher**
(This is a rebased and expanded version of #6015 now that the self-assembling switch zone has landed.)

I need to make some changes to zone-setup to support Reconfigurator rebalancing boundary NTP zones, and found it a little trickier to work with than expected. I think there were two reasons for this:

* The use of clap's builder interface instead of derive (addressed in https://github.com/oxidecomputer/omicron/commit/3708774aefde0aa6abcb8d12321c993f6268b4ae), which requires a surprising amount of boilerplate and custom parsing functions, most of which are now gone
* Using Result<(), CmdError::Failure> throughout, when really only main cares about that and everywhere else could use anyhow::Result<()> (mostly addressed in https://github.com/oxidecomputer/omicron/commit/cac26ddcd3e2345780d6f29863263b0611396f24); I also tried to make sure we always attach some .context() to errors (the old code did this in some cases but not all)

I also made a cleanup pass through the new switch zone user setup code; the biggest win there was reusing `illumos_utils::execute()` to check for success and build detailed error messages on failure. There are some other minor changes like long flags with different names (AFAICT we only ever use the short flags in SMF manifests, so this shouldn't require other changes) and potential bugfixes (e.g., the switch zone setup code was only correct for 0 or 1 profile, and should now be correct for 2+ if we ever need more than one).

While all these _should_ be correct, I want to run this through madrid before merging. I think it's ready for review, though; will update here once I do the madrid test.

### CODE REVIEW

File: zone-setup/src/bin/zone-setup.rs
Comment (karencfv) [inline zone-setup/src/bin/zone-setup.rs:485]: Nit:

```suggestion
                "failed to create chrony config file at: config_path {}",
```

File: zone-setup/src/switch_zone_user.rs
Comment (karencfv) [inline zone-setup/src/switch_zone_user.rs:54]: Oooh! This is much better!

**Review by karencfv — APPROVED** [review]
Thanks a bunch for taking the time to improve this code, it's looking a whole lot nicer! Just left an optional nit, but otherwise looks great to me!

I do agree that it's best to test out on Madrid before merging, but I'll leave it approved so you can merge once the testing is done.

**Comment by jgallagher** [comment]
Tested on `london`; that did catch one small problem that blocked correct configuration of `wicketd` (fixed in [788669e](https://github.com/oxidecomputer/omicron/pull/6175/commits/788669ec06618ca45a2eee8d99893fa356d29fda)). As of that commit, I went through:

* clean install
* RSS
* mupdate

all successfully.

**Pull request merged**
